engibench/problems/photonics2d/v0.py

```diff
@@ -422,32 +414,37 @@ def optimize(  # noqa: PLR0915
         # --- Get the frame saving interval from conditions for plotting ---
         save_frame_interval = conditions.get("save_frame_interval", 0)
 
-        # --- Initial Simulation for Normalization Constants E01, E02 ---
-        print("Optimize: Calculating E01/E02 using initial design...")  # Keep this info message
+        # --- Initial Simulation for first OptiStep history (IOG calculation) ---
+        print("Optimize: Calculating initial design...")  # Keep this info message
         epsr_init, ez1_init, ez2_init, source1_init, source2_init, probe1_init, probe2_init = self._run_fdfd(
             starting_point, conditions
         )
 
-        self._E01 = npa.abs(npa.sum(npa.conj(ez1_init) * probe1_init)) * 1e6
-        self._E02 = npa.abs(npa.sum(npa.conj(ez2_init) * probe2_init)) * 1e6
-
-        if self._E01 == 0 or self._E02 == 0:
-            print(f"Warning: Initial overlap zero (E01={self._E01:.3e}, E02={self._E02:.3e}). Using fallback.")
-            self._E01 = self._E01 if self._E01 != 0 else 1e-9
-            self._E02 = self._E02 if self._E02 != 0 else 1e-9
-
         self._source1 = source1_init
         self._source2 = source2_init
         self._probe1 = probe1_init
         self._probe2 = probe2_init
+        # Calculate overlaps
+        initial_overlap1 = mode_overlap(ez1_init, self._probe1)
+        initial_overlap2 = mode_overlap(ez2_init, self._probe2)
+        total_overlap_initial = initial_overlap1 * initial_overlap2
+        # Calculate initial material penalty
+        initial_penalty = penalty_weight * np.linalg.norm(starting_point)
+
+        # Add initial design performance into opti_steps_history
+        self.opti_steps_history: list[OptiStep] = []
```

Comment:
Why is this a class attribute now? I guess we can leave this as a local variable no?

Reply:
Ah, I was using this for debugging if I stopped the run early (e.g., interrupted the `optimize` solve before it was complete, so that I could diagnose some bugs), but we can move this back to a local variable now. In general, I could imagine it being useful to others to be able to get intermediate results of the optisteps if `optimize` fails for some reason during convergence, but for now we can make this local.

Reply:
This is fixed now. Should be able to merge I think @ffelten